docs/evaluation/how-to-evals/evaluating-phoenix-traces.md

```diff
@@ -0,0 +1,196 @@
+
+
+```python
+from phoenix.otel import register
+
+tracer_provider = register(project_name="evaluating_traces_quickstart")
+```
+
+## Prepare trace dataset
+
+For the sake of making this guide fully runnable, we'll briefly generate some traces and track them in Phoenix. Typically, you would have already captured traces in Phoenix and would skip to "Download trace dataset from Phoenix"
+
+
+```bash
+%%bash
+pip install -q openinference-instrumentation-openai
+```
+
+
+```python
+from openinference.instrumentation.openai import OpenAIInstrumentor
+
+OpenAIInstrumentor().instrument(tracer_provider=tracer_provider)
+```
+
+
+```python
+from openai import OpenAI
+
+# Initialize OpenAI client
+client = OpenAI()
+
+
+# Function to generate a joke
+def generate_joke():
+    response = client.chat.completions.create(
+        model="gpt-3.5-turbo",
+        messages=[
+            {"role": "system", "content": "You are a helpful assistant that generates jokes."},
+            {"role": "user", "content": "Tell me a joke."},
+        ],
+    )
+    joke = response.choices[0].message.content
+    return joke
+
+
+# Generate 5 different jokes
+jokes = []
+for _ in range(5):
+    joke = generate_joke()
+    jokes.append(joke)
+    print(f"Joke {len(jokes)}:\n{joke}\n")
+
+print(f"Generated {len(jokes)} jokes and tracked them in Phoenix.")
+```
+
+## Download trace dataset from Phoenix
+
+
+```python
+import phoenix as px
+
+spans_df = px.Client().get_spans_dataframe(project_name="evaluating_traces_quickstart")
+spans_df.head()
+```
+
+## Generate evaluations
+
+Now that we have our trace dataset, we can generate evaluations for each trace. Evaluations can be generated in many different ways. Ultimately, we want to end up with a set of labels and/or scores for our traces.
+
+You can generate evaluations using:
+- Plain code
+- Phoenix's [built-in LLM as a Judge evaluators](https://docs.arize.com/phoenix/evaluation/how-to-evals/running-pre-tested-evals)
+- Your own [custom LLM as a Judge evaluator](https://docs.arize.com/phoenix/evaluation/how-to-evals/bring-your-own-evaluator)
+- Other evaluation packages
+
+As long as you format your evaluation results properly, you can upload them to Phoenix and visualize them in the UI.
+
+Let's start with a simple example of generating evaluations using plain code. OpenAI has a habit of repeating jokes, so we'll generate evaluations to label whether a joke is a repeat of a previous joke.
+
+
+```python
+# Create a new DataFrame with selected columns
+eval_df = spans_df[["context.span_id", "attributes.llm.output_messages"]].copy()
+eval_df.set_index("context.span_id", inplace=True)
+
+# Create a list to store unique jokes
+unique_jokes = set()
+
+
+# Function to check if a joke is a duplicate
+def is_duplicate(joke_data):
+    joke = joke_data[0]["message.content"]
+    if joke in unique_jokes:
+        return True
+    else:
+        unique_jokes.add(joke)
+        return False
```

Comment:
I think this evaluation process might be slightly too hard to relate to -  I wonder if we can find a better "simple" eval